src/Workspaces/Core/Portable/Storage/SQLite/Interop/NativeMethods.cs

```diff
@@ -0,0 +1,172 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using SQLitePCL;
+
+namespace Microsoft.CodeAnalysis.SQLite.Interop
+{
+    [SuppressMessage("Style", "IDE1006:Naming Styles", Justification = "Name chosen to match SQLitePCL.raw")]
+    internal static class NativeMethods
+    {
+        public static SafeSqliteHandle sqlite3_open_v2(string filename, int flags, string vfs, out Result result)
+        {
+            result = (Result)raw.sqlite3_open_v2(filename, out var wrapper, flags, vfs);
+            if (result != Result.OK)
+            {
+                wrapper = null;
```

Comment:
What is the type of var here?  (Example of why I so dislike such use of var... makes it very hard to do effective code reviews.)  After searching around, it looks like it's a `sqlite3?`?  This line is setting it to null, which suggests result might not be OK but wrapper might still be non-null?  In that case, might something still need to be freed?

Reply:
`sqlite_open_v2` always returns a non-null `wrapper`, but it doesn't seem to need to be disposed unless the result is OK. I don't like this pattern one bit but the P/Invoke layer exposes a non-standard API so I'm trying to make use of what's available.